tinygrad/codegen/cstyle.py

```diff
@@ -24,6 +24,8 @@ class CStyleLanguage(NamedTuple):
   float4: Optional[str] = None
   half_prekernel: Optional[str] = None
   double_prekernel: Optional[str] = None
+  need_kernel_launcher: bool = False
+  build_kernel_launcher: Optional[Callable] = None
```

Comment:
This shouldn't be part of cstyle, it should be part of HIPProgramCPU

Reply:
Let me explained more for this and for build_kernel_launcher:
For HIP GPU backend, we just call hiprtcCompileProgram to build the kernel  and call hipModuleLaunchKernel, it works well.
HIP CPU backend has no HIP RTC support, so we need to find another approaches.
I tried building the kernel into a dynamic library, export it as function and call this function using hipModuleLaunchKernel, it doesn't work, hipModuleLaunchKernel can't access the function since they are from different HIP CPU contexts.
build_kernel_launcher is for fixing this issue, it will append calls to hipLaunchKernelGGL/hipStreamSynchronize after the kernel, so that all the code are compiled together by HIPProgramCPU into one library, and a special function named launch_and_wait_KERNEL_NAME_PLACEHOLDER is exported for launching the kernel.
Can we fix this without modifying cstyle.py?